packages/docusaurus-theme-classic/src/theme/TOC/index.tsx

```diff
@@ -42,18 +38,60 @@ export function TOCHeadings({
             // eslint-disable-next-line react/no-danger
             dangerouslySetInnerHTML={{__html: heading.value}}
           />
-          <TOCHeadings isChild toc={heading.children} />
+          <TOCHeadings
+            isChild
+            toc={heading.children}
+            maxHeadingLevel={maxHeadingLevel}
+            minHeadingLevel={minHeadingLevel}
+          />
         </li>
-      ))}
-    </ul>
-  );
+      );
+      // if we're not at the min level yet AND we have children at future levels, don't
+      // wrap the recursive `TOCHeadings` component with another <ul>
+    } else if (heading.level < minHeadingLevel && heading.children) {
```

Comment:
why not filter the headings that are < min ahead of time? 

That would prevent having to deal with such use-case and we wouldn't need the extra isInnerList prop?

Reply:
Seems legit. I didn't think of that!